test/e2e/specs/test.spec.js

```diff
@@ -16,46 +16,53 @@ describe('Vaults UI Test Cases', () => {
           walletName: 'user1',
         });
       } else {
-        const walletAddress = {
-          value: null,
-        };
         cy.setupWallet({
           createNewWallet: true,
           walletName: 'my created wallet',
+          selectedChains: ['Agoric'],
         });
-        cy.origin('https://wallet.agoric.app/', () => {
-          cy.visit('/wallet/');
 
-          cy.get('input[type="checkbox"]').click();
-          cy.contains('Proceed').click();
-        });
-        cy.acceptAccess();
-
-        cy.origin('https://wallet.agoric.app/', () => {
-          cy.visit('/wallet/');
+        cy.getWalletAddress('Agoric').then(address => {
+          // provision BLD
+          cy.request({
+            method: 'POST',
+            url: 'https://emerynet.faucet.agoric.net/go',
+            body: {
+              address,
+              command: 'delegate',
+              clientType: 'SMART_WALLET',
+            },
+            headers: {
+              Accept:
+                'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7',
+              'Content-Type': 'application/x-www-form-urlencoded',
+            },
+            timeout: 4 * MINUTE_MS,
+            retryOnStatusCodeFailure: true,
+          }).then(resp => {
+            expect(resp.body).to.eq('success');
+          });
 
-          cy.contains(/agoric.{39}/).spread(element => {
-            return element.innerHTML.match(/agoric.{39}/)[0];
+          // provision IST
+          cy.request({
+            method: 'POST',
+            url: 'https://emerynet.faucet.agoric.net/go',
+            body: {
+              address,
+              command: 'client',
+              clientType: 'SMART_WALLET',
+            },
+            headers: {
+              Accept:
+                'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7',
+              'Content-Type': 'application/x-www-form-urlencoded',
+            },
+            timeout: 4 * MINUTE_MS,
+            retryOnStatusCodeFailure: true,
+          }).then(resp => {
+            expect(resp.body).to.eq('success');
```

Comment:
Given the duplication in the code, I think you should create a function or a custom command taking `command` as an argument.